mne_bids/write.py

```diff
@@ -861,6 +865,9 @@ def write_raw_bids(raw, bids_basename, output_path, events_data=None,
 
     raw = raw.copy()
 
+    if anonymize:
+        raw = raw.anonymize()
```

Comment:
this will not work for formats that are not `fif`

Reply:
Ok, so I think almost there but I'm getting the below error when saving after anonymizing. Maybe @bloyl knows why raw.annotations.orig_time got set to a tuple? It is causing issues with save.

    ...: raw.save(raw_fname)                                                    
Writing /Users/alexrockhill/software/mne-bids/anonymize_tmp2-raw.fif
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-17-de48ed323732> in <module>
      1 raw_fname = 'anonymize_tmp2-raw.fif'
----> 2 raw.save(raw_fname)

</Users/alexrockhill/software/mne-python/mne/externals/decorator.py:decorator-gen-147> in save(self, fname, picks, tmin, tmax, buffer_size_sec, drop_small_buffer, proj, fmt, overwrite, split_size, split_naming, verbose)

~/software/mne-python/mne/utils/_logging.py in wrapper(*args, **kwargs)
     88             with use_log_level(verbose_level):
     89                 return function(*args, **kwargs)
---> 90         return function(*args, **kwargs)
     91     return FunctionMaker.create(
     92         function, 'return decfunc(%(signature)s)',

~/software/mne-python/mne/io/base.py in save(self, fname, picks, tmin, tmax, buffer_size_sec, drop_small_buffer, proj, fmt, overwrite, split_size, split_naming, verbose)
   1549         _write_raw(fname, self, info, picks, fmt, data_type, reset_range,
   1550                    start, stop, buffer_size, projector, drop_small_buffer,
-> 1551                    split_size, split_naming, part_idx, None, overwrite)
   1552 
   1553     @copy_function_doc_to_method_doc(plot_raw)

~/software/mne-python/mne/io/base.py in _write_raw(fname, raw, info, picks, fmt, data_type, reset_range, start, stop, buffer_size, projector, drop_small_buffer, split_size, split_naming, part_idx, prev_fname, overwrite)
   1957     if lasts[-1] > stop:
   1958         lasts[-1] = stop
-> 1959     sk_onsets, sk_ends = _annotations_starts_stops(raw, 'bad_acq_skip')
   1960     do_skips = False
   1961     if len(sk_onsets) > 0:

~/software/mne-python/mne/annotations.py in _annotations_starts_stops(raw, kinds, name, invert)
    483         # onsets are already sorted
    484         onsets = raw.annotations.onset[idxs]
--> 485         onsets = _sync_onset(raw, onsets)
    486         ends = onsets + raw.annotations.duration[idxs]
    487         onsets = raw.time_as_index(onsets, use_rounding=True)

~/software/mne-python/mne/annotations.py in _sync_onset(raw, onset, inverse)
    458     else:
    459         offset = -raw._first_time if inverse else raw._first_time
--> 460         annot_start = (raw.annotations.orig_time - meas_date) - offset + onset
    461     return annot_start
    462 

TypeError: unsupported operand type(s) for -: 'tuple' and 'float'

In [18]: raw.annotations.orig_time                                              
Out[18]: (946684800, 0)

In [19]: data_path = op.join(base_path, 'brainvision', 'tests', 'data') 
    ...: raw_fname = op.join(data_path, 'test.vhdr')                            

In [20]:                                                                        

In [20]: raw2 = mne.io.read_raw_brainvision(raw_fname)                          
Extracting parameters from /Users/alexrockhill/software/mne-python/mne/io/brainvision/tests/data/test.vhdr...
Setting channel info structure...

In [21]:                                                                        

In [21]: raw2.annotations                                                       
Out[21]: <Annotations  |  14 segments : New Segment/ (1), Stimulus/S253 (2), Stimulus/S255 (5)..., orig_time : 2013-11-13 16:14:03.794231>

In [22]: raw2.annotations.orig_time                                             
Out[22]: 1384359243.794231

raw2 is the original vhdr BV file and raw is the anonymized version.

Reply:
@alexrockhill when you ask for help with code, remember to format your message correctly. I started doing it but ran out of time. If you can finish it, I think it would help @bloyl to give feedback

Reply:
My bad, I had to run to do something else. Thanks for cleaning it up.

Reply:
That's my bad.

Just opened https://github.com/mne-tools/mne-python/pull/7016 to fix it.